frontera/contrib/backends/remote/messagebus.py

```diff
@@ -94,18 +117,20 @@ def states(self):
 
 
 def aggregate_per_host(requests):
-    per_host = dict()
-    for link in requests:
-        if b'fingerprint' not in link.meta[b'domain']:
+    per_host = defaultdict(list)
+
+    for request in requests:
+        try:
+            key = request.meta[b'domain'][b'fingerprint']
+            per_host[key].append(request)
+        except KeyError:
             continue
-        host_fprint = link.meta[b'domain'][b'fingerprint']
-        if host_fprint not in per_host:
-            per_host[host_fprint] = []
-        per_host[host_fprint].append(link)
+
     return per_host
 
 
 def get_host_fprint(request):
-    if b'fingerprint' not in request.meta[b'domain']:
+    try:
+        return request.meta[b'domain'][b'fingerprint']
+    except KeyError:
```

Comment:
Side note:
I think we should hard-code UrlFingerprintMiddleware and DomainMiddleware during initialization in Manager. Without them Frontera isn't operational. So it doesn't make any sense to check for presence of this data, or check if these middlewares are enabled.